crypto/ml_dsa/ml_dsa_key.c

```diff
@@ -139,13 +209,15 @@ ML_DSA_KEY *ossl_ml_dsa_key_dup(const ML_DSA_KEY *src, int selection)
                         vector_copy(&ret->s2, &src->s2);
                         vector_copy(&ret->t0, &src->t0);
                     }
-                    if (src->priv_encoding != NULL) {
-                        if ((ret->priv_encoding =
-                                OPENSSL_memdup(src->priv_encoding,
-                                               src->params->sk_len)) == NULL)
-                            goto err;
-                    }
+                    if ((ret->priv_encoding =
+                            OPENSSL_memdup(src->priv_encoding,
+                                           src->params->sk_len)) == NULL)
+                        goto err;
                 }
+                if (src->seed != NULL
+                    && (ret->seed = OPENSSL_memdup(src->priv_encoding,
```

Comment:
src->priv_encoding? (Doesnt look like it is hitting this path when testing)

Reply:
Good catch. Dup of keys that already have key material is rather rare at the EVP layer. So this slipped through.
Thanks!